src/BenchmarkDotNet/Diagnosers/DiagnosersLoader.cs

```diff
@@ -96,10 +97,16 @@ private static IDiagnoser[] LoadClassic()
 
         private static Assembly LoadDiagnosticsAssembly(Assembly benchmarkDotNetCoreAssembly)
         {
+            // We couldn't use Assembly.GetEntryAssembly()?.GetReferencedAssemblies() because
             // it not enough to just install NuGet to be "referenced", the project has to consume the dll for real to be on the referenced assembly list
-            var referencedAssemblyName = Assembly.GetEntryAssembly()?.GetReferencedAssemblies().SingleOrDefault(name => name.Name == DiagnosticAssemblyName);
-            if (referencedAssemblyName != default(AssemblyName))
-                return Assembly.Load(referencedAssemblyName);
+            // So we use DependencyContext that reads *.deps.json file that contains all referenced assembly. 
+            var inCompileLibraries = DependencyContext.Default.CompileLibraries.Any(l => l.Name.Equals(DiagnosticAssemblyName, StringComparison.OrdinalIgnoreCase));
```

Comment:
You shouldn't check the `CompileLibraries` at all. These are not appropriate for runtime scenarios, and are mainly there to support things like ASP.NET MVC Razor, which compiles code at runtime using Roslyn.